### PR TITLE
Documentation about vendoring

### PR DESCRIPTION
**Pull request by LK4D4**
ping @erikh @unclejack

### CODE REVIEW

**Comment by erikh** [comment]
Why can't we make the restore and save processes part of the makefile instead?

File: HACKING.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+* Save it in your project and rewrite imports:
+```
+godep save -r ./...
+```
+
+Update vendored packages
+========================
+
+Same as add, but you don't need to `go get` package.
```
Comment (mavenugo) [inline HACKING.md]: From my experience. godep save -r ./... kinda screws up updates.
I had consistent success with explicit **godep update {package name}** for updating a given package followed by **godep save -r ./...** to make sure the updated packages also gets the recursive godep vendored dependencies rewritten.

**Comment by dave-tucker** [comment]
Adding to @mavenugo's comments on updates, while `godep save -r ./...` is helpful after the update to make sure the paths get rewritten, it's probably worth adding a warning to check the `git diff` and make sure you've picked up the right changes before you commit. Too often have I meant to update one dependency and accidentally updated another dependency from my GOPATH that I'd been hacking on in parallel. Doesn't really matter if you have a clean GOPATH though.

**Comment by LK4D4** [comment]
@erikh I have no strong reasons, but seems wrong put to `Makefile` `GOPATH`-dependent commands. You can add it later if you wish, but vendor updating is not very often operation to deserve more targets for make I think.

**Comment by LK4D4** [comment]
@dave-tucker thanks!

**Comment by LK4D4** [comment]
@dave-tucker I added warning.

File: HACKING.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+Add vendored packages
+=====================
+
+For updating vendored package you need docker-network in your `GOPATH`. It's
+probably better to use clean `GOPATH` to avoid interferring with forks and other
```
Comment (mrunalp) [inline HACKING.md]: typo: interferring -> interfering

File: HACKING.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+```
+go get github.com/example/example
+```
+
+* Move to new package directory:
+```
+$GOPATH/src/github.com/example/example
+```
+
+* Checkout to desired version:
```
Comment (mrunalp) [inline HACKING.md]: Maybe "Checkout the desired version" might sound better?

**Comment by erikh** [comment]
@LK4D4 can I close this and we revisit it together tomorrow?

**Comment by LK4D4** [comment]
@erikh Yes, you can

**Pull request closed without merging**